src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/service/ExtractionService.java

```diff
@@ -53,7 +57,17 @@ public void register(ExtractionContainer container) {
   }
 
   public void extract() {
-    extractor.extract();
+    List<ExtractionContainer> extracted = extractor.extract();
+    for (ExtractionContainer container : extracted) {
+      runActions(container);
+    }
+  }
+
+  private void runActions(ExtractionContainer extractionContainer) {
+    List<PostExtractionAction> actions = extractionContainer.getPostExtractionActions();
+    for (PostExtractionAction action : actions) {
+      action.run();
```

Comment:
what if any of those actions threw an exception do you want to stop, or continue with the next action?

Reply:
Hmm that's a good point. If `DeleteTableAction` fails, I think it's acceptable to continue because it doesn't affect the replication, but just the cleanup of BigQuery temporary tables.
If the schema update fails, the table/partition will be unreadable. So it should stop replicating and cleanup the GCS bucket.

I'll need make it do what I want and add some tests to make sure this will always happen.

Reply:
Probably the easiest for now is to leave this code as it is - i.e. it will fail "fast" and not try any other actions if one of them fails. Then in each specific action it can decide whether to try/catch exceptions and not throw any if they don't want to stop the process and fail the job (which we definitely want if the schema update fails).

Reply:
I can add a try-catch in `DeleteTableAction` and log any errors that occur. Otherwise, the current logic _should_ make it fail if any error occurs in those actions.

Reply:
Sure, I guess the question is, would we ever want a failure in `DeleteTableAction` to stop the whole process? If not then what you suggest is fine. If we don't know for certain at present because we don't know enough about possible failures that could happen in that code, then I'd say leave it as is for now and we revisit this if it becomes an issue later.

Reply:
Yeah, probably better to leave it for later and see what the feedback is from users as well.